src/index.ts

```diff
@@ -7,14 +7,56 @@ import { isTestMode } from "./utils/env_helper";
 import { UIInteractionParams, UIInteraction } from "./events/ui_interaction";
 import { buildContexts } from "./utils/context_helper";
 
+const hash = require("object-hash");
+
 const snowplow = require("snowplow-tracker");
 const emitter = snowplow.emitter;
 const tracker = snowplow.tracker;
 const swdcTracker = <any>{};
+const ONE_HOUR_MILLIS = 1000 * 60 * 60;
 
 let lastProcessedTestEvent: any = {};
+let outgoingCodetimeEventMap: any = {};
+let pendingCodetimeEventTimer: any = undefined;
+
+const outgoingEventReconciler = (body: any) => {
```

Comment:
does the `body` have a specific type we should use or is `any` the right choice here?

Reply:
its either undefined if its an error or an js object. I'll check on the type though

Reply:
its basically a self object of request.js